src/metatensor/models/cli/eval_model.py

```diff
@@ -147,12 +160,23 @@ def eval_model(
         fileformat=options["structures"]["file_format"],
         dtype=dtype,
     )
+
+    # Calculate neighbor lists
+    for structure in eval_structures:
+        attach_neighbor_lists(structure, model.requested_neighbors_lists())
```

Comment:
Yeah, I think now it's much better. Thank you a lot, @frostedoyster ! 
Suggestion: shall we unify the _neighbor_ vs. _neighbors_ naming usage? I think we may have ambiguous naming of the NLs in different parts of the code.

Also _attaching_ the NL in some sense is the same as _adding_ the NL, which is the method of the System. Maybe `get_system_with_neighbors_lists()` will be better?

Reply:
`compute_required_neighbors_lists(structure, model)` could also work

Reply:
In this case such a function should return the computed NL (or a list of NLs) IMO, no the structure with the the NLs attached